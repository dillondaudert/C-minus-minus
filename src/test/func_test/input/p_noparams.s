	.section	.rodata
.int_wformat: .string "%d\n"
.str_wformat: .string "%s\n"
.int_rformat: .string "%d"
	.comm _gp, 16, 4

	.globl a1
	.type a1 @function
	.globl a2
	.type a2 @function
	.globl a3
	.type a3 @function
	.globl a4
	.type a4 @function
	.globl main
	.type main @function

a1:	nop
	pushq	%rbp
	movq	%rsp, %rbp
a2:	nop
	pushq	%rbp
	movq	%rsp, %rbp
a3:	nop
	pushq	%rbp
	movq	%rsp, %rbp
a4:	nop
	pushq	%rbp
	movq	%rsp, %rbp
main:	nop
	pushq	%rbp
	movq	%rsp, %rbp
	ret
	ret
	ret
	ret
	leave
	ret
	.size	a1, .-a1
	.size	a2, .-a2
	.size	a3, .-a3
	.size	a4, .-a4
	.size	main, .-main

