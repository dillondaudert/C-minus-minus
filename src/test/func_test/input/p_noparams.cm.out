st_init created the symbol table hash map
Added i to $$names, count=1
Adding new symbol: i, _gp, offset: 0, type: 0, size: 0, arr: 0
st_add added i at position 72
Added j to $$names, count=2
Adding new symbol: j, _gp, offset: 1, type: 0, size: 0, arr: 0
st_add added j at position 75
Added k to $$names, count=3
Adding new symbol: k, _gp, offset: 2, type: 0, size: 0, arr: 0
st_add added k at position 74
Added l to $$names, count=4
Adding new symbol: l, _gp, offset: 3, type: 0, size: 0, arr: 0
st_add added l at position 77
updating symbol i
Symbol 72:i found
Symbol i updated with size 4, offset 0
updating symbol j
Symbol 75:j found
Symbol j updated with size 4, offset 4
updating symbol k
Symbol 74:k found
Symbol k updated with size 4, offset 8
updating symbol l
Symbol 77:l found
Symbol l updated with size 4, offset 12



Adding new symbol: a1, a1, offset: 0, type: 2, size: 4, arr: 0
st_add added a1 at position 0
Added process a1 to stable





Adding new symbol: a2, a2, offset: 0, type: 2, size: 4, arr: 0
st_add added a2 at position 192
Added process a2 to stable






Adding new symbol: a3, a3, offset: 0, type: 2, size: 4, arr: 0
st_add added a3 at position 128
Added process a3 to stable







Adding new symbol: a4, a4, offset: 0, type: 2, size: 4, arr: 0
st_add added a4 at position 320
Added process a4 to stable









Adding new symbol: main, main, offset: 0, type: 2, size: 4, arr: 0
st_add, added main to collision LL at 0
Added process main to stable









Static: 	.comm _gp, 16, 4

Freed symbol at position 0
Freed LL symbol at position 0
Freed link in collision LL at 0
Freed symbol at position 72
Freed symbol at position 74
Freed symbol at position 75
Freed symbol at position 77
Freed symbol at position 128
Freed symbol at position 192
Freed symbol at position 320
Freed symbol table
